src/components/components/AddLayerPanel/AddLayerPanel.component.jsx

```diff
@@ -0,0 +1,362 @@
+      handlerFunction: createStreetmixStreet
+    },
+    {
+      name: 'Entity from extruded SVG',
+      img: '',
+      icon: '',
+      description:
+        'Create entity with svg-extruder component, that accepts a svgString and creates a new entity with geometry extruded from the svg and applies the default mixin material grass.',
+      id: 3,
+      handlerFunction: createSvgExtrudedEntity
+    },
+    {
+      name: '3D Tiles',
+      img: '',
+      icon: '',
+      description:
+        'Adds an entity to load and display 3d tiles from Google Maps Tiles API 3D Tiles endpoint. This will break your scene and you cannot save it yet, so beware before testing.',
+      id: 4,
+      handlerFunction: create3DTiles
+    },
+    {
+      name: 'Create custom model',
+      img: '',
+      icon: '',
+      description:
+        'Create entity with model from path for a glTF (or Glb) file hosted on any publicly accessible HTTP server.',
+      id: 5,
+      handlerFunction: createCustomModel
+    },
+    {
+      name: 'Create primitive geometry',
+      img: '',
+      icon: '',
+      description:
+        'Create entity with A-Frame primitive geometry. Geometry type could be changed in properties panel.',
+      id: 6,
+      handlerFunction: createPrimitiveGeometry
+    }
+  ];
+
+  // get array with objects data (cardsData) from mixinGroups of selectedOption
+  const getSelectedMixinCards = (selectedOption) => {
+    if (!selectedOption) return [];
+    const selectedOptionData = options.find(
+      (option) => option.value === selectedOption
+    );
+    const selectedMixinGroupNames = selectedOptionData.mixinGroups;
+
+    // there are no mixin groups
+    if (!selectedMixinGroupNames) return [];
+
+    // filter selected mixin groups from all mixin groups (groupedMixins)
+    const cardsData = groupedMixins
+      .filter((group) => selectedMixinGroupNames.includes(group.label))
+      .flatMap((mixinGroup) => mixinGroup.options);
+
+    return cardsData;
+  };
+
+  let selectedCards;
+  if (selectedOption == 'Layers: Streets & Intersections') {
+    selectedCards = layersData;
+  } else {
+    selectedCards = getSelectedMixinCards(selectedOption);
+  }
+
+  const handleSelect = (value) => {
+    setSelectedOption(value);
+  };
+
+  /* create and preview entity events */
+
+  // entity preview element
+  let preEntity = document.createElement('a-entity');
+  let cameraFrontVec = new THREE.Vector3();
+  let selectedObjPos = new THREE.Vector3();
+  let segmentElevationPosY = 0;
+  let segmentOfSelectedEl;
+
+  preEntity.setAttribute('visible', false);
+
+  AFRAME.scenes[0].appendChild(preEntity);
+
+  const previewEntity = () => {
+    if (selectedOption == 'Models: Buildings') {
+      preEntity.setAttribute('scale', { x: 0.1, y: 0.1, z: 0.1 });
+    } else {
+      preEntity.setAttribute('scale', { x: 0.5, y: 0.5, z: 0.5 });
+    }
+    /* place object in front of active camera */
+    const direction = new THREE.Vector3();
```

Comment:
You can move direction variable out of the function so it's not created every time.